src/porepy/numerics/ad/equation_system.py

```diff
@@ -1029,37 +997,46 @@ def projection_to(self, variables: Optional[VariableList] = None) -> sps.csr_mat
             return sps.csr_matrix((0, num_dofs))
 
     def dofs_of(self, variables: VariableList) -> np.ndarray:
-        """Get the indices in the global vector of unknowns belonging to the variable(s).
+        """Get the indices in the global vector of unknowns belonging to the variables.
 
         Parameters:
             variables: VariableType input for which the indices are requested.
 
         Returns:
-            an order-preserving array of indices of DOFs belonging to the VariableType input.
+            An array of indices/ DOFs corresponding to ``variables``.
+            Note that the order of indices corresponds to the order in ``variables``.
 
         Raises:
-            ValueError: if unknown VariableType arguments are passed.
+            ValueError: If an unknown  variable is passed as argument.
 
         """
         variables = self._parse_variable_type(variables)
         global_variable_dofs = np.hstack((0, np.cumsum(self._variable_num_dofs)))
 
-        # Storage of indices per requested variable.
-        indices = list()
-        for variable in variables:
-            var_number = self._variable_numbers[variable]
-            var_indices = np.arange(
-                global_variable_dofs[var_number],
-                global_variable_dofs[var_number + 1],
-                dtype=int,
-            )
-            indices.append(var_indices)
+        indices: list[np.ndarray] = []
+
+        for var in variables:
```

Comment:
@keileg I reverted the changes for `dofs_of`. Order corresponds now again to input.
Validation was added that variable is known to equation system.